src/components/Navbar/TransactionsList.tsx

```diff
@@ -0,0 +1,172 @@
+  } = useContext(ThemeContext)
+  const [isExpanded, setIsExpanded] = useState<boolean>()
+
+  const TxStatusElem = (status: TxStatus) => {
+    switch (status) {
+      case TxStatus.pending:
+        return <Loading size="small" />
+      case TxStatus.success:
+        return (
+          <div style={{ color: colors.icon.success }}>
+            <CheckCircleOutlined />
+          </div>
+        )
+      case TxStatus.failed:
+        return (
+          <div style={{ color: colors.icon.failure }}>
+            <ExclamationCircleOutlined />
+          </div>
+        )
+    }
+  }
+
+  const hasPendingTxs = useMemo(
+    () => transactions?.some(tx => tx.status === TxStatus.pending),
+    [transactions],
+  )
+
+  useEffect(() => {
+    // Auto expand if pending tx is added
+    if (hasPendingTxs) setIsExpanded(true)
+  }, [hasPendingTxs])
+
+  return (
+    <div style={{ ...style }}>
+      {!!transactions?.length && (
+        <div
+          className="clickable-border"
+          role="button"
+          style={{
+            cursor: 'pointer',
+            display: 'flex',
+            alignItems: 'center',
+            paddingLeft: 10,
+            paddingRight: 5,
+            borderRadius: 15,
+            height: 30,
+          }}
+          onClick={() => setIsExpanded(!isExpanded)}
+        >
+          {hasPendingTxs ? <Loading size="small" /> : <ThunderboltOutlined />}
+          <span style={{ minWidth: 20, textAlign: 'center', fontWeight: 600 }}>
+            {isExpanded ? <CaretDownOutlined /> : transactions.length}
+          </span>
+        </div>
+      )}
+
+      {isExpanded && (
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 10,
+            width: 300,
+            ...listStyle,
+          }}
+        >
+          {transactions?.length ? (
+            transactions
+              .sort((a, b) =>
+                timestampForTxLog(a) > timestampForTxLog(b) ? -1 : 1,
+              )
+              .map(tx => (
+                <div
+                  key={tx.tx.hash}
+                  style={{
+                    display: 'flex',
+                    alignItems: 'center',
+                    justifyContent: 'space-between',
+                    gap: 10,
+                    width: '100%',
+                    padding: '5px 20px',
+                    background: colors.background.l0,
+                    border: `1px solid ${colors.stroke.secondary}`,
+                    boxSizing: 'border-box',
+                  }}
+                >
+                  <a
+                    style={{
+                      cursor: tx.tx.hash ? 'pointer' : 'default',
+                      color: colors.text.primary,
+                      fontSize: '0.7rem',
+                      lineHeight: '1.5rem',
+                    }}
+                    href={
+                      tx.tx.hash
+                        ? `https://${readNetwork.name}${
```

Comment:
might be nice to kick this out to a utility function, getEtherscanTxLink or something like that

Reply:
yes. luv